iotfunctions/pipeline.py

```diff
@@ -2342,30 +2350,51 @@ def _execute_stage(self, stage, df, start_ts, end_ts, entities, register, to_csv
         self.trace_add(msg, created_by=stage, df=newdf)
         return newdf
 
-    def apply_scope(self, df, stage):
-        if hasattr(stage, 'scope') and stage.scope is not None:
-            logger.debug('Applying Scope')
+    def apply_scope(self, df, scope):
+        has_scope = False
+        scope_mask = None
+        if scope:
             eval_expression = ''
-            scope = json.loads(stage.scope)
+            has_scope = True
             if scope.get('type') == 'DIMENSIONS':
                 logger.debug('Applying Dimensions Scope')
                 dimension_count = len(scope.get('dimensions'))
                 for dimension_filter in scope.get('dimensions'):
                     dimension_name = dimension_filter['dimension_name']
```

Comment:
I think this property should be name and value

Reply:
I went ahead with `dimension_name` as UI was already using it

Reply:
@herleraja Done